File: js/datastructures/Heap/PriorityQueue.test.js
const { assert, measurePerformance } = require('../../Utils');
const { PriorityQueue } = require('./PriorityQueue');

const pq = new PriorityQueue();

pq.add(4);
pq.add(50);
pq.add(7);
pq.add(55);
pq.add(90);
pq.add(87);
pq.add(2);

console.log('' + pq);
assert([2, 50, 4, 55, 90, 87, 7], pq.heap);

const removed = pq.remove();
console.log('removed: ' + removed);
console.log('' + pq);
assert([4, 50, 7, 55, 90, 87], pq.heap);

File: java/datastructures/src/test/java/org/datastructures/PriorityQueueTest.java
package org.datastructures;

import org.junit.jupiter.api.Test;

public class PriorityQueueTest {

  @Test
  public void testPriorityQueue() {
    PriorityQueue thePQ = new PriorityQueue(5);
    thePQ.insert(30);
    thePQ.insert(50);
    thePQ.insert(10);
    thePQ.insert(40);
    thePQ.insert(20);

    while (!thePQ.isEmpty()) {
      long item = thePQ.remove();
      assert item == 10 || item == 20 || item == 30 || item == 40 || item == 50;
      System.out.print(item + " "); // 10, 20, 30, 40, 50
    }
  }
}
